### PR TITLE
It hurts to lose your groin.

### LINKED ISSUE #12884: Eat groin without consequences
**Issue Description**:
You can eat a person's groin with matter eater without it affecting them.
They still lose their organs but they don't take any damage from the being eaten part.

**What did you expect to happen**: 
I ate my own groin and expected to die.

**What happened instead**:
I didn't die but just lost my organs which did not really impact me either.

**Why is this bad/What are the consequences**:
I guess no more killing people by eating their groins.

**Steps to reproduce the problem**:
1. get matter eater
2. aim at groin
3. chomp

**When did the problem start happening**:
I think a month ago, I don't recall.

**Extra information**:
Vox can't love.

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
Fixes #12884.

## Why It's Good For The Game
ERP bad. Punish eating groins.

## Testing
Injected matter eater.
Ate own groin.
OH GOD THAT WAS A BAD IDEA! WHY DID I DO THAT?
image

## Changelog
:cl:
fix: Eating someone's groin with Matter Eater now does massive damage, rather than basically nothing.
/:cl:

### CODE REVIEW

File: code/game/dna/mutations/mutation_powers.dm [hunk 1]
```diff
@@ -428,7 +428,27 @@
 				return
 			user.visible_message("<span class='danger'>[user] [pick("chomps","bites")] off [the_item]'s [limb]!</span>")
 			playsound(user.loc, 'sound/items/eatfood.ogg', 50, 0)
-			limb.droplimb(0, DROPLIMB_SHARP)
+			if(istype(limb,/obj/item/organ/external/groin))
+				limb.receive_damage(100, sharp=TRUE)
```
Comment (DGamerL) [inline code/game/dna/mutations/mutation_powers.dm:432]:
```suggestion
				limb.receive_damage(100, sharp = TRUE)
```

File: code/game/dna/mutations/mutation_powers.dm [hunk 2]
```diff
@@ -428,7 +428,27 @@
...
+
+				var/obj/item/organ/external/right_leg = H.get_organ(BODY_ZONE_R_LEG)
+				if(istype(right_leg))
+					right_leg.droplimb(FALSE, DROPLIMB_SHARP)
+
+				var/obj/item/organ/external/chest = H.get_organ(BODY_ZONE_CHEST)
+				if(istype(chest))
+					chest.receive_damage(50, sharp = TRUE)
+			else
+				limb.droplimb(FALSE, DROPLIMB_SHARP)
```
Comment (DGamerL) [inline code/game/dna/mutations/mutation_powers.dm:449]: Could you maybe invert this if-statement?

Comment (FunnyMan3595) [reply code/game/dna/mutations/mutation_powers.dm]: Yup, that worked. Done.

File: code/game/dna/mutations/mutation_powers.dm [hunk 3]
```diff
@@ -428,7 +428,26 @@
...
+
+				var/obj/item/organ/external/right_leg = H.get_organ(BODY_ZONE_R_LEG)
+				if(istype(right_leg))
+					right_leg.droplimb(FALSE, DROPLIMB_SHARP)
+
+				var/obj/item/organ/external/chest = H.get_organ(BODY_ZONE_CHEST)
+				if(istype(chest))
+					chest.receive_damage(50, sharp = TRUE)
+
+
```
Comment (DGamerL) [inline code/game/dna/mutations/mutation_powers.dm:450]:
```suggestion
```

**Review by DGamerL — APPROVED** [review]
Code LGTM

File: code/game/dna/mutations/mutation_powers.dm [hunk 4]
```diff
@@ -428,7 +428,25 @@
 				return
 			user.visible_message("<span class='danger'>[user] [pick("chomps","bites")] off [the_item]'s [limb]!</span>")
 			playsound(user.loc, 'sound/items/eatfood.ogg', 50, 0)
-			limb.droplimb(0, DROPLIMB_SHARP)
+
+			// Most limbs will drop here. Groin won't, but this
+			// still spills out the organs that were in it.
+			limb.droplimb(FALSE, DROPLIMB_SHARP)
+			if(istype(limb,/obj/item/organ/external/groin))
```
Comment (lewcc) [inline code/game/dna/mutations/mutation_powers.dm:435]:
```suggestion
			if(istype(limb, /obj/item/organ/external/groin))
```

**Comment by lewcc** [comment]
Failing due to https://github.com/ParadiseSS13/Paradise/pull/24569, PR is good.

**Review by lewcc — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]
Please do a merge master aswell

**Pull request merged**